spec/rubocop/cop/rspec/capybara/match_style_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::MatchStyle, :config do
+  it 'registers an offense when using `assert_style`' do
+    expect_offense(<<~RUBY)
+      page.find(:css, '#first').assert_style(display: 'block')
+                                ^^^^^^^^^^^^ Use `assert_matches_style` instead of `assert_style`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      page.find(:css, '#first').assert_matches_style(display: 'block')
+    RUBY
+  end
+
+  it 'registers an offense when using `has_style?`' do
+    expect_offense(<<~RUBY)
+      expect(page.find(:css, 'first')
+        .has_style?(display: 'block')).to be true
+         ^^^^^^^^^^ Use `matches_style?` instead of `has_style?`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      expect(page.find(:css, 'first')
+        .matches_style?(display: 'block')).to be true
```

Comment:
It seems we have two resulting styles depending on the original:
```ruby
expect(page.matches_style?(display: 'block')).to be true
```
and see in the next example:
```ruby
expect(page).to match_style(display: 'block')
```

`RSpec/PredicateMatcher` won't be able to detect it, will it?

Reply:
I opened a Pull Request for follow-up on this issue: https://github.com/rubocop/rubocop-rspec/pull/1495

Reply:
Please accept my apologies for not making myself clear.
What I meant is:
```diff
- expect(page.has_style?(display: 'block')).to be(true)
+ expect(page.matches_style?(display: 'block')).to be(true)
```
but
```diff
- expect(page).to have_style(display: 'block')
+ expect(page).to match_style(display: 'block')
```

While all those expectations are different notations of the same thing.

It doesn't feel natural to off-load this to `RSpec/PredicateMatcher`, so would this cop be able to suggest the same correction for various inputs?

Reply:
Offloading to `RSpec/PredicateMatcher` is certainly unnatural, but what about offloading to `RSpec/Capybara/PredicateMatcher`?
- https://github.com/rubocop/rubocop-rspec/pull/1495